networkx/algorithms/community/louvain.py

```diff
@@ -0,0 +1,531 @@
+    """
+    node2com = {}
+    total_weight = 0
+    internals = {}
+    degrees = {}
+    gdegrees = {}
+
+    def __init__(self) :
+        self.node2com = dict([])
+        self.total_weight = 0
+        self.degrees = dict([])
+        self.gdegrees = dict([])
+        self.internals = dict([])
+        self.loops = dict([])
+
+    def __str__(self) :
+        return ("node2com : " + str(self.node2com) + " degrees : "
+            + str(self.degrees) + " internals : " + str(self.internals)
+            + " total_weight : " + str(self.total_weight))
+
+    def copy(self) :
+        """Perform a deep copy of status"""
+        new_status = Status()
+        new_status.node2com = self.node2com.copy()
+        new_status.internals = self.internals.copy()
+        new_status.degrees = self.degrees.copy()
+        new_status.gdegrees = self.gdegrees.copy()
+        new_status.total_weight = self.total_weight
+
+    def init(self, graph, part = None) :
+        """Initialize the status of a graph with every node in one community"""
+        count = 0
+        self.node2com = dict([])
+        self.total_weight = 0
+        self.degrees = dict([])
+        self.gdegrees = dict([])
+        self.internals = dict([])
+        self.total_weight = graph.size(weight = 'weight')
+        if part == None :
+            for node in graph.nodes() :
+                self.node2com[node] = count
+                deg = float(graph.degree(node, weight = 'weight'))
+                if deg < 0 :
+                    raise ValueError("Bad graph type, use positive weights")
+                self.degrees[count] = deg
+                self.gdegrees[node] = deg
+                self.loops[node] = float(graph.get_edge_data(node, node,
+                                                 {"weight":0}).get("weight", 1))
+                self.internals[count] = self.loops[node]
+                count = count + 1
+        else :
+            for node in graph.nodes() :
+                com = part[node]
+                self.node2com[node] = com
+                deg = float(graph.degree(node, weight = 'weight'))
+                self.degrees[com] = self.degrees.get(com, 0) + deg
+                self.gdegrees[node] = deg
+                inc = 0.
+                for neighbor, datas in graph[node].items() :
+                    weight = datas.get("weight", 1)
+                    if weight <= 0 :
+                        raise ValueError("Bad graph type, use positive weights")
+                    if part[neighbor] == com :
+                        if neighbor == node :
+                            inc += float(weight)
+                        else :
+                            inc += float(weight) / 2.
+                self.internals[com] = self.internals.get(com, 0) + inc
+
+
+
+def __neighcom(node, graph, status) :
+    """
+    Compute the communities in the neighborood of node in the graph given
+    with the decomposition node2com
+    """
+    weights = {}
```

Comment:
Use `weights = defaultdict(lambda: 0)` so you can just use `weights[neighborcom] += weight` below.